webui/react/src/pages/InteractiveTask.tsx

```diff
@@ -49,11 +49,23 @@ const getTitleState = (commandState?: CommandState, taskName?: string): string =
 
 export const InteractiveTask: React.FC = () => {
   const [pageView, setPageView] = useState<PageView>(PageView.IFRAME);
-  const { taskId, taskName, taskResourcePool, taskUrl, taskType } = useParams<Params>();
+  const {
+    taskId: tId,
+    taskName: tName,
+    taskResourcePool: tResourcePool,
+    taskType: tType,
+    taskUrl: tUrl,
+  } = useParams<Params>();
   const [taskState, setTaskState] = useState<CommandState>();
   const storeDispatch = useStoreDispatch();
   const { ui } = useStore();
 
+  const taskId = tId ?? '';
+  const taskName = tName ?? '';
+  const taskResourcePool = tResourcePool ?? '';
```

Comment:
nit: Not necessarily specific to this body of work but I think I would personally rather us show some sort of "error no task Id" etc page instead of defaulting to a blank string. This extends to all of the query parameters that are required for a component to be rendered correctly that we will now just replace with `' '`. I don't think it should be in this PR but I would be happy with it being a ticket in the backlog to refactor these.

Reply:
In react router dom v6, `usePram` returns `type | undefined` even though we are sure the param never be undefined. These code wouldn't be `undefined`, but  I needed to use `??`, `assertion` or `as` to make it work.  For now, I use `??` for safety. 
I agree with your opinion though. We should find a way to handle it well.

Reply:
For the future, if there is a code path that you know should never be hit you should throw an exception. I usually define an `InvariantFailed` exception for this purpose.